doc_source/resource-type-walkthrough.md

```diff
@@ -1085,6 +1065,190 @@ We'll also need to update the unit test for the delete handler\.
    }
    ```
 
+#### Code the Read Handler<a name="resource-type-walkthrough-implement-read-handler-code"></a>
+
+1. In your IDE, open the `ReadHandler.java` file, located in the `src/main/java/com/example/testing/wordpress` folder\.
+
+1. Replace the entire contents of the `ReadHandler.java` file with the following code\.
+
+   ```
+   package com.example.testing.wordpress;
+
+   import com.amazonaws.services.ec2.AmazonEC2;
+   import com.amazonaws.services.ec2.AmazonEC2ClientBuilder;
+   import com.amazonaws.services.ec2.model.Instance;
+   import software.amazon.cloudformation.proxy.*;
+
+   public class ReadHandler extends CustomBaseHandler {
+
+       private static final String DELETED_INSTANCE_STATE = "terminated";
+
+       @Override
+       public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+           final AmazonWebServicesClientProxy proxy,
+           final ResourceHandlerRequest<ResourceModel> request,
+           final CallbackContext callbackContext,
+           final Logger logger) {
+
+           final ResourceModel model = request.getDesiredResourceState();
+
+           AmazonEC2 ec2Client = AmazonEC2ClientBuilder.standard().withRegion(SUPPORTED_REGION).build();
+
+           Instance currentInstanceState = retrieveCurrentInstanceState(proxy, ec2Client, model.getInstanceId());
+
+           if (currentInstanceState.getInstanceId() == null || DELETED_INSTANCE_STATE.equals(currentInstanceState.getState().getName())) {
```

Comment:
The read handler must return a Not Found error if the instance has been deleted.